src/triage/triage.service.ts

```diff
@@ -1,63 +1,482 @@
+  ) {}
+
+  async handleNewAlert(
+    dto: SubmitAlertDto,
+    userId: string,
+    tenantId: string,
+    source: string,
+  ) {
+    // Determine the alert type (txtp)
+    const txtp =
+      typeof dto?.result?.transaction?.TxTp === 'string'
+        ? dto.result.transaction.TxTp
+        : '';
+
+    try {
+      const newAlert = await this.prisma.alert.create({
+        data: {
+          tenant_id: tenantId,
+          priority: Priority.LOW,
+          source: source,
+          txtp: txtp,
+          alert_status: AlertStatus.NEW,
+          message: String(dto.result.message),
+          alert_data: dto.result.report,
+          transaction: dto.result.transaction,
+          network_map: dto.result.networkMap,
+          confidence_per: 0,
+        },
+      });
+      await this.audit.logAction({
+        userId,
+        operation: 'ALERT_CREATED',
+        entityName: 'Alert',
+        actionPerformed: `Created new alert ${newAlert.alert_id}`,
+        outcome: 'SUCCESS',
+      });
+
+      return newAlert;
+    } catch (error) {
+      this.logger.error('Error creating alert', error);
+      throw new InternalServerErrorException('Failed to create alert');
+    }
+  }
+
+  async updateAlertData(
+    alertId: string,
+    dto: UpdateAlertDto,
+    userId: string,
+    tenantId: string,
+  ) {
+    const existingAlert = await this.prisma.alert.findUnique({
+      where: { alert_id: alertId },
+    });
+
+    if (!existingAlert) {
+      throw new NotFoundException(`Alert ${alertId} not found`);
+    }
+
+    if (existingAlert.tenant_id !== tenantId) {
+      throw new NotFoundException(
+        `Alert ${alertId} not accessible for this tenant`,
+      );
+    }
+
+    if (existingAlert.alert_status === AlertStatus.CLOSED) {
+      throw new BadRequestException(
+        `Alert ${alertId} is closed status and can not be updated`,
+      );
+    }
+
+    try {
+      const updatedAlert = await this.prisma.alert.update({
+        where: { alert_id: alertId },
+        data: {
+          confidence_per: dto.confidence_per,
+          priority: dto.priority,
+        },
+      });
+
+      await this.audit.logAction({
+        userId,
+        operation: 'ALERT_UPDATED',
+        entityName: 'Alert',
+        actionPerformed:
+          `Updated alert ${alertId}` +
+          (dto.confidence_per !== undefined
+            ? `, confidence_per=${dto.confidence_per}`
+            : '') +
+          (dto.priority !== undefined ? `, priority=${dto.priority}` : ''),
+        outcome: 'SUCCESS',
+      });
+
+      return updatedAlert;
+    } catch (error) {
+      this.logger.error(`Update failed for alert ${alertId}`, error);
+      throw new InternalServerErrorException('Failed to update alert');
+    }
+  }
+
+  async manualCloseAlert(
+    alertId: string,
+    closeAlertDto: CloseAlertDto,
+    userId: string,
+    tenantId: string,
+  ) {
+    const alert = await this.prisma.alert.findUnique({
+      where: { alert_id: alertId },
```

Comment:
Fetching using tenantId here might be better instead of having a conditional check later in the code.
```suggestion
      where: { alert_id: alertId, tenantId: tenantId },
```

Reply:
Updated alert fetching by using both alertId and tenantId directly